server/tserver/src/main/java/org/apache/accumulo/tserver/ScanServer.java

```diff
@@ -149,8 +149,8 @@ private TabletMetadataLoader(Ample ample) {
         loadAll(Set<? extends KeyExtent> keys) {
       long t1 = System.currentTimeMillis();
       @SuppressWarnings("unchecked")
-      var tms = ample.readTablets().forTablets((Collection<KeyExtent>) keys).build().stream()
-          .collect(Collectors.toMap(tm -> tm.getExtent(), tm -> tm));
+      var tms = ample.readTablets().forTablets((Collection<KeyExtent>) keys, e -> {}).build()
```

Comment:
Seems like we use a NOOP consumer in a few places. Might be useful to assign that to a variable like `ignoreNotFound`

Reply:
I would like to to pass something that lets ample know they don't care about the missing extents, because the implementation could then stream directly from the scanner.  When the caller wants to know what is missing ample has to buffer everything.  Thought about passing in an Optional<Consumer<>>, then when a caller does not care they can pass in Optional.empty().

Reply:
Yeah, I'm just suggesting giving it a name, so it's more clear, rather than have the lambda be anonymous.

For example:

```java

  private static final Consumer<KeyExtent> IGNORE_MISSING = e -> {};
  // ...

      var tms = ample.readTablets().forTablets((Collection<KeyExtent>) keys, IGNORE_MISSING).build()
```

Reply:
I made it optional in 0533f92.  Its not used frequently so the object creation cost does not matter.

Reply:
I still think the named function is easier to understand, and doesn't require the unnecessary wrapping/unwrapping of a Consumer into an Optional. While not important for performance reasons, I just don't think the boilerplate wrapping/unwrapping is necessary or very clean. There is plenty of precedent for naming functions for reuse like this, either by assigning to a variable or by using a method reference. I just think it's cleaner, and makes the code much easier to read. But it's not that important to me (not enough to put in the effort to persuade, anyway), and I guess the Optional does  achieve a similar purpose.

Reply:
I am not trying to avoid the cost of creating a lambda.  The implementation needs to know when the user does not care about the missing extents so it can take a much more efficient code path.  With only a lambda there is no way to know if the consumer will discard everything its passed or not.  That is why the implementation needs to differentiate between a lambda being passed and one not being passed.

Reply:
I think I understand what you're saying now. My comment was only about readability (giving `e -> {}` a name). I now understand that you had entirely unrelated reasons for choosing Optional (to bypass doing unnecessary costly tasks). If I'm understanding correctly and the use of Optional allows you to bypass those costly tasks, then I think Optional makes sense. It just wasn't clear to me that the discussion had moved away from the readability point I was making.

Reply:
> If I'm understanding correctly and the use of Optional allows you to bypass those costly tasks, then I think Optional makes sense.

That's correct.  There was javadoc covering this, but it was not very detailed.  I added more details to the java doc in 958b868